spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/AbstractConnectionFactory.java

```diff
@@ -84,20 +94,24 @@ public void handleRecovery(Recoverable recoverable) {
 
 	};
 
-	private volatile ExecutorService executorService;
+	private ExecutorService executorService;
```

Comment:
Just curious, why remove volatile?

Reply:
The properties from setters are considered as configuration options which mostly are specified only during application start up. They are not intended to be changed concurrently at runtime.
And even if they may be changed at runtime concurrently that is not so critical do not get an actual value immediately in other thread.
Without `volatile` we get some performance gain. 

Those properties which are really critical for runtime changes remain with `volatile`. For example `active`, `stopped`, `running` etc.

I do such a refactoring only in the effected code though and only when I remember 😄